pennylane/operation.py

```diff
@@ -468,15 +468,27 @@ def __init__(self, *params, wires=None, do_queue=True, id=None):
         if wires is None:
             raise ValueError(f"Must specify the wires that {self.name} acts on")
 
-        self._num_params = len(params)
-        # Check if the expected number of parameters coincides with the one received.
-        # This is always true for the default `Operator.num_params` property, but
-        # subclasses may overwrite it to define a fixed expected value.
-        if len(params) != self.num_params:
-            raise ValueError(
-                f"{self.name}: wrong number of parameters. "
-                f"{len(params)} parameters passed, {self.num_params} expected."
-            )
+        # Check whther the expected number of parameters (in case the Operation subclass defines it via the num_params property) coincides with the one received.
+        if hasattr(self.__class__, "num_params"):
+            # pylint: disable=no-member, access-member-before-definition
+            if (
+                isinstance(self.__class__, int)
+                and len(params) != self.__class__.num_params
```

Comment:
I am blind, what is this catching? How can the op class be an int?

Reply:
Uh! This is a typo! It should be `isinstance(self.__class__.num_params, int)`. The logic schold be as follows:
If a class sets e.g. `num_params = 4` then `self.__class__.num_params` is an `int` and we want an error if `len(params)` does not match. If instead the class defined an `@property` decorated method (because `num_params` may depend on the instance) than `self.__class__.num_params` is of type `property`, we compare self.num_params with `len(params)`. Thinking of it now: We may want to move this test after setting `self._wires` because in such cases num_params may depend on `num_wires` (this happens for "template" like Operations). I will push a fix asap.